File: src/js/4._queue.js
// Estructura de datos
/** Queue */
// Array usando como cola FIFO

/** @type {string} */
const cola = [];

// agregamos al final (enqueue)
cola.push('Melissa');
cola.push('Luis');
cola.push('Valeria');

console.log(cola);

// Sacar del principio (dequeue)
/** @type {string} */
let siguiente = cola.shift();
console.log(siguiente);
console.log(cola);

// siguiente en cola
siguiente = cola.shift();
console.log(siguiente);
console.log(cola);

File: src/python/4._queue.py
# Estructura de datos
# Queue (Cola)

queue: list[str] = []

# agregamos al final (enqueue)
queue.append("pikachu")
queue.append("Bulbasaur")
queue.append("Charmander")

print(queue)

# sacar del principio
siguiente: str = queue.pop(0)
print(f"Siguiente en salir: {siguiente}")
print(queue)

# siguiente en cola
siguiente = queue.pop(0)
print(f"Siguiente en salir: {siguiente}")
print(queue)